lib/src/common/disposable.dart

```diff
@@ -284,6 +300,14 @@ class Disposable implements _Disposable, DisposableManagerV3 {
       _logger.info(
           '$runtimeType $hashCode took ${stopwatch.elapsedMicroseconds / 1000000.0} seconds to dispose');
     }
+
+    flagLeak(runtimeType.toString());
+  }
+
+  @mustCallSuper
+  @override
+  void flagLeak(String description) {
+    _leakFlag = new LeakFlag(description);
```

Comment:
Do we expect consumers to decorate other classes with the `LeakFlagger` mixin, or do we only expect it to be used in `Disposable`? If we only expect `Disposable` to use it, I'd be inclined to not bother with the `LeakFlagger` mixin at all and then just make this a private method.